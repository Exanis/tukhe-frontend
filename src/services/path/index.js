export function api(path) {
    const basePath = process.env.REACT_APP_API_URL;

    return basePath + path;
}

export function oauthUrl() {
    return process.env.REACT_APP_OAUTH_URL;
}

/*
 * The next two functions are either copied or adapted from satellizer's code
 * https://github.com/sahat/satellizer/blob/master/src/utils.ts
 */

export function getFullUrlPath(location) {
  const isHttps = location.protocol === 'https:';
  return location.protocol + '//' + location.hostname +
    ':' + (location.port || (isHttps ? '443' : '80')) +
    (/^\//.test(location.pathname) ? location.pathname : '/' + location.pathname);
}

export function parseQueryString(str) {
    const result = {};

    (str || '').split('&').map(keyValue => {
        if (keyValue) {
            const parts = keyValue.split('=');
            const key = decodeURIComponent(parts[0]);

            if (parts[1] === undefined)
                result[key] = true;
            else
                result[key] = decodeURIComponent(parts[1]);
        }
        return '';
    });
    return result;
}