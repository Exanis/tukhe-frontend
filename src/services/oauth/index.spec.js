import {
    oauth1Popup,
    pollOauth1Popup
} from "./index";
import * as path from '../path';

it('Should open popup as expected', () => {
    window.open = jest.fn();

    expect(oauth1Popup(
        'testurl',
        'token',
        300,
        200
    )).toMatchSnapshot();
    expect(window.open).toHaveBeenCalled();
    expect(window.open.mock.calls[0]).toMatchSnapshot();
});

it('Should correctly poll oauth popup', () => {
    expect(pollOauth1Popup(false)).toBeFalsy();
    expect(pollOauth1Popup({
        test: 'test'
    })).toBeFalsy();
    expect(pollOauth1Popup({
        closed: true
    })).toBeFalsy();
    expect(pollOauth1Popup({
        closed: false,
        location: 'something'
    })).toBeTruthy();
    jest.spyOn(path, 'getFullUrlPath');
    jest.spyOn(path, 'oauthUrl');
    path.getFullUrlPath.mockReturnValue(false);
    path.oauthUrl.mockReturnValue(false);
    expect(pollOauth1Popup({
        closed: false,
        location: false
    })).toBeTruthy(); // It should not fail here
    expect(path.getFullUrlPath).toHaveBeenCalledWith(false);
    jest.spyOn(path, 'parseQueryString');
    path.parseQueryString.mockReturnValue({
        error: true
    });
    expect(pollOauth1Popup({
        closed: false,
        location: {
            search: 'test',
            hash: 'test'
        }
    })).toBeFalsy();
    path.parseQueryString.mockReturnValue({});

    const fn = jest.fn();

    expect(pollOauth1Popup({
        closed: false,
        location: {
            search: 'test',
            hash: 'test'
        },
        close: fn
    })).toEqual({});
    expect(fn).toHaveBeenCalled();
});
