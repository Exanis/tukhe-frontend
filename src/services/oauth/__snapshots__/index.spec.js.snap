// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should open popup as expected 1`] = `undefined`;

exports[`Should open popup as expected 2`] = `
Array [
  "testurl?oauth_token=token",
  "popupWindow",
  "width=200,height=300,top=187.2,left=412",
]
`;
